nixos/modules/services/misc/omnom.nix

```diff
@@ -0,0 +1,178 @@
+        description = "The omnom service user.";
+      };
+
+      group = lib.mkOption {
+        type = lib.types.nonEmptyStr;
+        default = "omnom";
+        description = "The omnom service group.";
+      };
+
+      settings = lib.mkOption {
+        description = ''
+          Configuration options for the /etc/omnom/config.yml file.
+        '';
+        type = lib.types.submodule {
+          freeformType = settingsFormat.type;
+          options = {
+            app = {
+              debug = lib.mkEnableOption "the omnom debug mode.";
+              disable_signup = lib.mkEnableOption "restricting user creation.";
+              results_per_page = lib.mkOption {
+                type = lib.types.int;
+                default = 20;
+                description = "Number of results per page.";
+              };
+            };
+            db = {
+              connection = lib.mkOption {
+                type = lib.types.str;
+                default = "${cfg.dataDir}/db.sqlite3";
+                description = "Database connection URI.";
+                defaultText = lib.literalExpression ''
+                  "''${config.services.omnom.dataDir}/db.sqlite3"
+                '';
+              };
+              type = lib.mkOption {
+                type = lib.types.str;
+                default = "sqlite";
+                description = "Database type.";
+              };
+            };
+            server = {
+              address = lib.mkOption {
+                type = lib.types.str;
+                default = "127.0.0.1:7331";
+                description = "Server address.";
+              };
+              secure_cookie = lib.mkOption {
+                type = lib.types.bool;
+                default = true;
+                description = "Limit cookies to secure channel.";
+              };
+            };
+            storage = {
+              root = lib.mkOption {
+                type = lib.types.str;
+                default = "${cfg.dataDir}/static/data";
+                defaultText = lib.literalExpression ''
+                  "''${config.services.omnom.dataDir}/static/data"
+                '';
+                description = "Where the snapshots are saved.";
+              };
+            };
+          };
+        };
+        default = { };
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    environment.etc."omnom/config.yml".source = pkgs.writers.writeYAML "omnom-config.yml" cfg.settings;
+
+    services.omnom.settings = {
+      app.debug = lib.mkDefault cfg.debug;
+      storage.type = lib.mkDefault "fs";
+    };
+
+    systemd.services.omnom = {
+      path = [ pkgs.rsync ];
+
+      serviceConfig = {
+        DynamicUser = isDynamicUser;
+        User = cfg.user;
+        Group = cfg.group;
+        ExecStart =
+          let
+            args = lib.cli.toGNUCommandLineShell { } {
+              config = "/etc/omnom/config.yml";
+              d = cfg.debug;
+            };
```

Comment:
This feels overkill. Why not just `--config /etc/omnom/config.yml ${lib.optionalString cfg.debug "-d"}`?

Reply:
I wanted to make things more readable, but I guess it's a bit too much for just two flags.

Which one seems cleaner to you?

```nix
ExecStart = "${lib.getExe cfg.package} listen --config /etc/omnom/config.yml ${lib.optionalString cfg.debug "-d"}";
```

or

```nix
ExecStart =
  "${lib.getExe cfg.package} listen --config /etc/omnom/config.yml "
  + (lib.optionalString cfg.debug "-d");
```